Source/DafnyCore/DooFile.cs

```diff
@@ -168,18 +169,21 @@ public class ManifestData {
     }
 
     return CheckAndGetLibraryOptions(reporter, file, options, origin, Manifest.Options,
-      new Dictionary<Option, OptionCompatibility.OptionCheck>());
+      ImmutableDictionary<Option, OptionCompatibility.OptionCheck>.Empty);
   }
 
   public static DafnyOptions? CheckAndGetLibraryOptions(ErrorReporter reporter,
     Uri libraryFile,
     DafnyOptions options, IToken origin,
     IDictionary<string, object> libraryOptions,
-    Dictionary<Option, OptionCompatibility.OptionCheck> additionalOptions) {
+    ImmutableDictionary<Option, OptionCompatibility.OptionCheck> additionalOptions) {
     var result = new DafnyOptions(options);
     var success = true;
+    var totalOptions = additionalOptions.Merge(OptionChecks,
+      (_, _) => throw new InvalidOperationException());
```

Comment:
For next opportunity: having a MergeDistinct variant would make this more readable (I had to go look up what the function value was for to understand)